File: C#/src/Intermediate/day44/Exercise.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Intermediate.day44
{
    public class Exercise
    {
        public static int ex1CheckBit(int A, int B)
        {
            return (A >> B) & 1;
        }
        public static int ex2NumberOf1Bits(int A)
        {
            int count = 0;
            while (A > 0)
            {
                if ((A & 1) == 1)
                {
                    count++;
                }
                A = A >> 1;
            }
            return count;
        }
        public static int ex3SetBit(int A, int B)
        {
            return (1 << A) | (1 << B);
        }
        public static int ex4UnSetBit(int A, int B)
        {
            if (((A >> B) & 1) == 1)
            {
                A ^= (1 << B);
            }
            return A;
        }
    }
}

File: Python/src/day44/Exercise.py
class Exercise:
    def ex1CheckBit(self, A, B):
        return (A>>B)&1

    def ex2NumberOf1Bits(self, A):
        count = 0
        while A>0:
            if A&1:
                count += 1
            A = A>>1
        return count

    def ex3SetBit(self, A, B):
        return (1<<A)|(1<<B)

    def ex4UnSetBit(self, A, B):
        if (A>>B)&1:
            return A^(1<<B)
        return A